modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs

```diff
@@ -191,5 +192,47 @@ public partial class Node
         {
             return GetParent() as T;
         }
+
+        /// <summary>Finds a <see cref="Node"/> of type T starting at the root of the tree. Returns <see langword="null"/> if none is found</summary>
+        /// <typeparam name="T">Any type which inherits from <see cref="Node"/></typeparam>
+        public T FindNodeOfType<T>() where T : Node
+        {
+            // find the root node
+            Node parent = this;
+            Node tempParent = parent;
+            while (tempParent != null)
+            {
+                tempParent = parent.GetParent();
+                if (tempParent != null)
+                {
+                    parent = tempParent;
+                }
+            }
+
+            return this.FindNodeOfType(parent);
+        }
+
+        /// <summary>Finds a <see cref="Node"/> of type T starting at the specified parent. Returns <see langword="null"/> if none is found</summary>
+        /// <param name="parent">A object that inherits from <see cref="Node"/></param>
+        /// <typeparam name="T">Any type which inherits from <see cref="Node"/></typeparam>
+        public T FindNodeOfType<T>(Node parent) where T : Node
```

Comment:
Why are there two methods? The first one says it starts at the root of the tree in the documentation but in reality it uses `this`, and the second method says it starts at the specified parent. These should be one method that mimics `find_node` in core.